### ISSUE TITLE
NumPy 2 prep

### ISSUE THREAD

**Issue by jstac** [QuantEcon/meta#116]
https://pythonspeed.com/articles/numpy-2/

**Comment by mmcky** [comment]
@oyamad has added `numpy 2` testing for QuantEcon.py and that is passing tests so looking good.

**Comment by oyamad** [comment]
Yes, the current QuantEcon.py is ready to release as a NumPy 2 compatible version!

**Comment by mmcky** [comment]
Thanks @oyamad 

I have released `quantecon=0.8.0`

https://github.com/QuantEcon/QuantEcon.py/releases/tag/v0.8.0

**Comment by mmcky** [comment]
Reading the initial link from @jstac 

- [ ] Upgrading your code with Ruff

should be run over lecture and QuantEcon.py code
